src/import_edf.jl

```diff
@@ -249,40 +251,103 @@ following transformations:
+following transformations:
+
+- the label is whitespace-stripped, parens-stripped, and lowercased
+- trailing generic EDF references (e.g. "ref", "ref2", etc.) are dropped
+- any instance of `+` is replaced with `_plus_` and `/` with `_over_`
+- all component names are converted to their "canonical names" when possible
+  (e.g. "m1" in an EEG-matched channel name will be converted to "a1").
 
 See the OndaEDF README for additional details regarding EDF formatting expectations.
 """
-function import_edf!(dataset::Dataset, edf::EDF.File, uuid::UUID=uuid4();
-                     custom_extractors=(), import_annotations::Bool=true)
+function edf_to_onda_samples(edf::EDF.File; custom_extractors=())
     EDF.read!(edf)
-    _, recording = create_recording!(dataset, uuid)
+    edf_samples = Samples[]
     for extractor in Iterators.flatten((STANDARD_EXTRACTORS, custom_extractors))
         extracted = extractor(edf)
         extracted === nothing && continue
-        onda_signal_name, onda_signal, edf_signals = extracted
-        samples = onda_samples_from_edf_signals(onda_signal, edf_signals, edf.header.seconds_per_record)
-        store!(dataset, uuid, onda_signal_name, samples)
+        samples_info, edf_signals = extracted
+        samples = onda_samples_from_edf_signals(samples_info, edf_signals, edf.header.seconds_per_record)
+        push!(edf_samples, samples)
     end
-    if import_annotations
-        for annotation_signal in edf.signals
-            annotation_signal isa EDF.AnnotationsSignal || continue
-            for record in annotation_signal.records
-                for tal in record
-                    start_nanosecond = Nanosecond(round(Int, 1e9 * tal.onset_in_seconds))
-                    if tal.duration_in_seconds === nothing
-                        stop_nanosecond = start_nanosecond
-                    else
-                        stop_nanosecond = start_nanosecond + Nanosecond(round(Int, 1e9 * tal.duration_in_seconds))
-                    end
-                    for annotation_string in tal.annotations
-                        isempty(annotation_string) && continue
-                        annotate!(recording, Annotation(annotation_string, start_nanosecond, stop_nanosecond))
-                    end
+    return edf_samples
+end
+
+"""
+    edf_to_onda_annotations(edf::EDF.File, uuid::UUID)
+
+Extract EDF+ annotations from an `EDF.File` for recording with ID `uuid` and
+return them as a vector of `Onda.Annotation`s.  If no annotations are found then
+an empty `Vector{Annotation}` is returned.
+"""
+function edf_to_onda_annotations(edf::EDF.File, uuid::UUID)
+    EDF.read!(edf)
+    annotations = Annotation[]
+    for annotation_signal in edf.signals
+        annotation_signal isa EDF.AnnotationsSignal || continue
+        for record in annotation_signal.records
+            for tal in record
+                start_nanosecond = Nanosecond(round(Int, 1e9 * tal.onset_in_seconds))
```

Comment:
I would construct a `Second` and then cast/round as necessary to avoid hard coded constants

Reply:
same as above